[Ji-just-ji] iP

ji-just-ji opened the pull request.

ji-just-ji committed:
Level 0
Initialize
Change name to Peko
Add intro and exit

ji-just-ji committed:
Fixed intro logo

ji-just-ji committed:
Level-1
Add echo functionality
Move introduction into separate function

ji-just-ji committed:
Add loop for echo

ji-just-ji committed:
Move final variables together

ji-just-ji committed:
Remove unneeded intitalizations

ji-just-ji committed:
Add Copypasta text
Add easteregg

ji-just-ji committed:
Update Copypasta text

ji-just-ji committed:
Updated Copypasta text

ji-just-ji committed:
Added read and write capability

ji-just-ji committed:
Modify input checking
Make echo its own function

ji-just-ji committed:
Fix write input

ji-just-ji committed:
Change to array to store to do list
Fix other functions to work with array

ji-just-ji committed:
Add Task object
Move tasks into Task object

ji-just-ji committed:
Add ToDo Deadline Event objects

ji-just-ji committed:
Add deadline timing capability

ji-just-ji committed:
Add deadline check

ji-just-ji committed:
Add InvalidTaskException to handle non task input
Update Task ToDos Deadline Event to throw InvalidTaskException

ji-just-ji committed:
Change Expected.txt header

ji-just-ji committed:
Setup automated text U/I testing
Add

ji-just-ji committed:
Fix task bugs
Fix automated test U/I testing bugs

ji-just-ji committed:
Create ACTUAL.TXT

ji-just-ji committed:
Fix runtest.sh changes

ji-just-ji committed:
Update List.txt

ji-just-ji committed:
Save attempt 1
Add SaveHandler

ji-just-ji committed:
Save attempt 1 cont

ji-just-ji committed:
Add save and load into List.txt

ji-just-ji committed:
Remove unused methods

ji-just-ji committed:
Update List.txt

ji-just-ji committed:
Merge branch 'Save_(Level-7)'

ji-just-ji committed:
Made to save upon every update

ji-just-ji committed:
Add datetime functionality

ji-just-ji committed:
Begin UIHandling

ji-just-ji committed:
More UIHandling

ji-just-ji committed:
Add echo and readArray

ji-just-ji committed:
Add mark and unmark

ji-just-ji committed:
Add todo deadline event

ji-just-ji committed:
UI handling draft1

ji-just-ji committed:
UI handling
Still has missing line breaks

ji-just-ji committed:
UI Handling bug fixes

ji-just-ji committed:
Fix easter egg

ji-just-ji committed:
Begin StorageHandler

ji-just-ji committed:
Fix some StorageHandler

ji-just-ji committed:
Make StorageHandler static

ji-just-ji committed:
Bug fixes

ji-just-ji committed:
Bug fixes

ji-just-ji committed:
Bug fixes

ji-just-ji committed:
Bug fixes

ji-just-ji committed:
Move things around
Add Parser UI TaskHandler
Moved Enum out

ji-just-ji committed:
Change StorageHandler to static

ji-just-ji committed:
Fix TaskHandler string name

ji-just-ji committed:
Update List.txt

ji-just-ji committed:
Merge branch 'StorageHandler'

ji-just-ji committed:
Move to all to peko package

ji-just-ji committed:
Bug fix

ji-just-ji committed:
Revert "Bug fix"

This reverts commit 7e170140e8366bac4811338f59779b6446f54b3f.

ji-just-ji committed:
Revert "Move to all to peko package"

This reverts commit 45d0c24e6f3bb0929a2957c830a207f995669014.

ji-just-ji committed:
Move to peko package

ji-just-ji committed:
Add autosave after every command

ji-just-ji committed:
Revert "Add autosave after every command"

This reverts commit d414291c1fadf703950abb87760dec8b14295155.

ji-just-ji committed:
Revert "Move to peko package"

This reverts commit 86e6be056807c9020b7e9de87374b5be6a0c167a.

ji-just-ji committed:
Bug fixes

ji-just-ji committed:
Fix not writing to List.txt

ji-just-ji committed:
Add build.gradle

ji-just-ji committed:
Add PekoTest for junit test

ji-just-ji committed:
Merge branch 'add-gradel-support-ji-just-ji'

ji-just-ji committed:
Add JavaDoc for Peko class

ji-just-ji committed:
Add JavaDoc for UserInputHandler class

ji-just-ji committed:
Add JavaDoc for StorageHandler class

ji-just-ji committed:
Add JavaDoc for SaveHandler class

ji-just-ji committed:
Add JavaDoc for DateTimeHandler class

ji-just-ji committed:
Add Find class with supporting methods

ji-just-ji committed:
Implement Find

ji-just-ji committed:
Fix switch case indents

ji-just-ji committed:
Fix some javadoc

ji-just-ji committed:
Merge pull request #1 from ji-just-ji/branch-Level-9

Branch level 9

ji-just-ji committed:
Merge branch 'master' into branch-A-JavaDoc

ji-just-ji committed:
Begin coding standard fix

ji-just-ji committed:
Begin Junit

ji-just-ji committed:
Create MANIFEST.MF

ji-just-ji committed:
Update README.md

anthonytamzil commented on src/main/java/peko/DateTimeHandler.java:
```diff
@@ -0,0 +1,37 @@
+import java.time.LocalDateTime;
+
+public class DateTimeHandler {
+    private LocalDateTime date;
+    private String dateTimeString;
+
+    private int year;
+    private int month;
+    private int day;
+    private int hour;
+    private int min;
+    public DateTimeHandler(String s) {
+
+        String[] dateTime = s.split(" ");
+        String[] dateString = dateTime[0].split("/");
+        int timeString = Integer.parseInt(dateTime[1]);
+        day = Integer.parseInt(dateString[0]);
+        month = Integer.parseInt(dateString[1]);
+        year = Integer.parseInt(dateString[2]);
+        hour = timeString/100;
+        min = timeString%100;
+        date = LocalDateTime.of(year,month,day,hour,min);
+        dateTimeString = date.getMonth() +  " " + date.getDayOfMonth() + " " + date.getYear();
+    }
```
Perhaps a couple of line breaks could be added in between to enhance readability?

anthonytamzil commented on src/main/java/peko/Task.java:
```diff
@@ -0,0 +1,34 @@
+
+public class Task {
+    protected String name;
+    protected boolean status;
```
Perhaps the boolean name can be changed to isDone instead of status?

anthonytamzil commented on src/main/java/peko/TaskHandler.java:
```diff
@@ -0,0 +1,131 @@
+        Task t = new Event(description);
+        StorageHandler.addToArray(t);
+        System.out.println(lineBreak);
+    }
```
Perhaps documentation can be added here especially where there is a lot of methods that may not be clear?

anthonytamzil commented on src/main/java/peko/TaskHandler.java:
```diff
@@ -0,0 +1,131 @@
+        StorageHandler.addToArray(t);
+        System.out.println(lineBreak);
+    }
+    private void Find(String s) {
```
Perhaps try to write the method names in camelCase?

anthonytamzil reviewed (commented):
Overall, code-wise looks good.
Perhaps can work a bit on the coding standard and line breaks.

ji-just-ji committed:
Load javafx

ji-just-ji committed:
Add screen and button press

ji-just-ji committed:
Add some GUI features

ji-just-ji committed:
Revert "Add some GUI features"

This reverts commit 1f4e8625eb24a4687c4a053bb1ddceb2f3f4f9c1.

lordidiot commented on src/main/java/peko/DateTimeHandler.java:
```diff
@@ -0,0 +1,37 @@
+
+import java.time.LocalDateTime;
+
+public class DateTimeHandler {
```
It would be good to have header comments for this public class.

lordidiot commented on src/main/java/peko/DateTimeHandler.java:
```diff
@@ -0,0 +1,37 @@
+        dateTimeString = date.getMonth() +  " " + date.getDayOfMonth() + " " + date.getYear();
+    }
+
+    public LocalDateTime getDate() {
```
re: header comments for public methods

lordidiot commented on src/main/java/peko/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+
+import java.util.Arrays;
+
+public class Deadline extends Task{
```
nit: Consider adding spacing before the `{`

lordidiot commented on src/main/java/peko/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+        }
+
+        String[] temp = split[0].split(" ",2);
+        //System.out.println(Arrays.toString(temp));
```
Consider removing your leftover testing code.

lordidiot commented on src/main/java/peko/Peko.java:
```diff
@@ -0,0 +1,30 @@
+package peko;
+
+import java.io.*;
```
Imported classes should be listed explicitly, consider changing the wildcard import

lordidiot commented on src/main/java/peko/Peko.java:
```diff
@@ -0,0 +1,30 @@
+    }
+    private void run() {
+        Output.intro();
+        while (true) {
```
I like this event loop

lordidiot commented on src/main/java/peko/UI.java:
Is this empty class used?

lordidiot reviewed (commented):
Overall looks great. Some common issues throughout include

* Lack of javadocs / comments for public methods
* Leftover testing code as comments
* Spacing and line break inconsistency

Otherwise it looks very fully featured!

ji-just-ji committed:
Fix attempt 1

ji-just-ji committed:
Revert "Revert "Add some GUI features""

This reverts commit 6f3462d9817f113b9299327fe12f68125067b637.

ji-just-ji committed:
Follow instructions

ji-just-ji committed:
Iteration 2

ji-just-ji committed:
Split DialogBox

ji-just-ji committed:
Split DialogBox

ji-just-ji committed:
Split DialogBoAdd return as string for GUI output

ji-just-ji committed:
Fix TaskHandler

ji-just-ji committed:
Finish TaskHandler fix

ji-just-ji committed:
Setup some responses in GUI

ji-just-ji committed:
Add flip for peko response

ji-just-ji committed:
Change gui header text

ji-just-ji committed:
Merge pull request #2 from ji-just-ji/GUI

GUI

ji-just-ji committed:
Merge branch 'master' into branch-A-JavaDoc

ji-just-ji committed:
Update javadocs for some classes

There are several classes lacking javadocs for
the class and certain methods.
Adding the javadocs to improve code quality and ease of understanding code.

ji-just-ji committed:
Update more javadocs for some classes

There are several classes lacking javadocs for
the class and certain methods.
Adding the javadocs to improve code quality and ease of understanding code.

ji-just-ji committed:
Update remaining javadocs for some classes

There are several classes lacking javadocs for
the class and certain methods.
Adding the javadocs to improve code quality and ease of understanding code.

ji-just-ji committed:
Fix double input bug

There was a bug that caused instructions to be executed twice.

ji-just-ji committed:
Add Junit Test for Deadline and Task classes

ji-just-ji committed:
Add missing assertions to junit tests

ji-just-ji committed:
Create ArchiveHandler

ji-just-ji committed:
Remove rewrite archive every time

ji-just-ji committed:
Override equals for Tasks to make sure identicle Tasks are based on task details.

ji-just-ji committed:
Add no duplicates in Archive functionality

ji-just-ji committed:
Bug fix for checking if exists in Archive already

ji-just-ji committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/List.txt
#	src/main/java/peko/Peko.java
#	src/main/java/peko/UserInputHandler.java
#	src/main/java/peko/memory/StorageHandler.java
#	src/main/java/peko/tasks/Event.java
#	src/main/java/peko/tasks/TaskHandler.java

ji-just-ji committed:
Added javadocs to ArchiveHandler and Peko

ji-just-ji committed:
Add archive option for user

ji-just-ji committed:
Revert "Add missing assertions to junit tests"

This reverts commit c1e6000eac7ddadfab94812b52c1dbd30f1fde4e.

ji-just-ji committed:
Add assertions for DeadlineTest

Checks if Deadline creates the appropriate deadlines with an input String.

ji-just-ji committed:
Merge pull request #4 from ji-just-ji/branch-A-Assertions

Add Assertions

ji-just-ji committed:
Remove unneeded functions in GUIController

ji-just-ji committed:
Remove unnecessary comments in Launcher

ji-just-ji committed:
Merge pull request #5 from ji-just-ji/GUI

Remove GUI branch

ji-just-ji committed:
Merge branch 'master' of https://github.com/ji-just-ji/ip

ji-just-ji committed:
Merge branch 'master' into branch-Level-10

ji-just-ji committed:
Merge pull request #6 from ji-just-ji/branch-Level-10

Merge branch level 10

ji-just-ji committed:
Commit some differences in line numbers and brackets

ji-just-ji committed:
Bug fix in Tasks

ji-just-ji committed:
Create new equals method for Tasks

Create new equals method such that tasks and Todos
are equal when their descriptions are equal
Create new equals method for Deadline such that
Deadline tasks are equal when their descriptions and
deadlines date and time are equal
Event tasks are equal when their descriptions, start date
and end date are equal.

ji-just-ji committed:
Add UI.png and Updated User Guide

ji-just-ji committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/peko/StorageHandler.java

ji-just-ji committed:
Rename UI to Ui
